zebra/rt_netlink.c

```diff
@@ -1229,6 +1330,27 @@ static bool _netlink_route_encode_nexthop_src(const struct nexthop *nexthop,
 	return true;
 }
 
+static size_t fill_seg6ipt_encap(char *buffer, size_t buflen,
+				 struct in6_addr *seg)
+{
+	struct seg6_iptunnel_encap *ipt;
+	struct ipv6_sr_hdr *srh;
+	const size_t srhlen = 24;
+
+	memset(buffer, 0, buflen);
+
+	ipt = (struct seg6_iptunnel_encap *)buffer;
+	ipt->mode = SEG6_IPTUN_MODE_ENCAP;
+	srh = ipt->srh;
+	srh->hdrlen = (srhlen >> 3) - 1;
+	srh->type = 4;
+	srh->segments_left = 0;
+	srh->first_segment = 0;
+	memcpy(&srh->segments[0], seg, sizeof(struct in6_addr));
```

Comment:
mmm - got to be more careful with buflen: it's not checked anywhere in here, and this code has to protect itself against a short buflen

Reply:
I added boundary check. and its future comment.
Currently, we don't support multiple-sid for Srv6-transit behavior. So we can represent really simple fixed boundary check, but we may face multiple sid. My comment is for that.

fixed on https://github.com/FRRouting/frr/pull/5865/commits/621f34868d4c14a96e4e7afb5c80af667233e6d5